ISI-407 ISI-411 Zusatzfelder in Abfragevariante sowie Abfragevarianten für Sachbearbeitung

boal opened the pull request:
**Description**

* Bei durch die Abfrageerstellung angelegten Abfragevarianten kann durch die Sachbearbeitung eine Ergänzung um zusätzliche  Informationen vorgenommen werden. Die Sachbearbietung kann keine bestehenden Felder ändern.
* Die Sachbearbeitung kann eigene Abfragevarianten anlegen. In diesen Abfragevarianten können alle Felder ausschließlich durch die Sachbearbeitung geändert werden.

**Reference**

ISI-407
ISI-411

lhmAschaenz committed:
ISI-407 Backend API Gen

lhmAschaenz committed:
ISI-407 Lookup Listen um sobonOrientierungswertJahr ergänzt

lhmAschaenz committed:
ISI-407 AbfragevarianteSachbearbeitungFormular mit Integration

lhmAschaenz committed:
ISI-407 AbfragevarianteSachbearbeitungFormular und API Gen

boal committed:
fix imports

boal committed:
add mapper

boal committed:
move mapper method from mixin to utils

boal committed:
fix possibly undifined

boal committed:
add patch for InBearbeitungSachbearbeitung

boal committed:
add contect for bearbeitung

boal committed:
move variables

boal committed:
add check for editable

boal committed:
add check for editable

lhmAschaenz committed:
ISI-407 Save work

boal committed:
add lodash method

boal committed:
adapt tree creation

lhmAschaenz committed:
Merge remote-tracking branch 'origin/ISI-411-Abfragevarianten-exklusiv-fuer-sachbearbeitung' into ISI-407-MVP-Status-Einblenden-von-Zusatzfelden-in-Abfragevariante

lhmAschaenz committed:
Merge branch ISI-411-Abfragevarianten-exklusiv-fuer-sachbearbeitung

boal committed:
adapt tree creation

boal committed:
adapt tree creation

boal committed:
adapt tree creation

boal committed:
adapt tree creation

boal committed:
add context

boal committed:
add context

boal committed:
remove unused variable

boal committed:
add context for abfragevariante

lhmAschaenz committed:
ISI-407 Bedarfsmeldungen Fachabteilungen

boal committed:
add context for abfragevariante

boal committed:
add abfragevarianteSachbearbeitung to factory

lhmAschaenz committed:
ISI-407 Bugfix open/close BedarfsmeldungFachabteilungenDialog

boal committed:
resolve merge conflicts

boal committed:
adapt validation for all abfragevarianten

boal committed:
adapt validation for all abfragevarianten

boal committed:
refactor is editable

boal committed:
refactor is editable

boal committed:
refactor is editable

boal committed:
refactor is editable

lhmAschaenz committed:
ISI-407 Implemented event handling Bedarfsmeldung

lhmAschaenz committed:
ISI-407 Validierung BedarfsmeldungFachabteilungen

boal committed:
remove duplicate

lhmAschaenz committed:
ISI-407 Merge ISI-411-Abfragevarianten-exklusiv-fuer-sachbearbeitung into branch

lhmAschaenz committed:
ISI407 Merge ISI-411-Abfragevarianten-exklusiv-fuer into branch

boal committed:
refaktor columns

lhmAschaenz committed:
ISI-407 Validierung und Steuerung Editierbarkeit

lhmAschaenz committed:
ISI-407 Bereinigung überflüssige Imports

lhmAschaenz committed:
ISI-407	Merge ISI-411-Abfragevarianten-exklusiv-fuer-sachbearbeitung' into branch

lhmAschaenz committed:
ISI-407

lhmAschaenz committed:
ISI-407 Bugfix SobonOrientierungswertJahr

lhmAschaenz committed:
ISI-407 Formatierung Bedarfsmeldungen

lhmAschaenz committed:
ISI-407 Bugfix Validierung Bedarfsmeldungsdialog

lhmAschaenz committed:
ISI-407

lhmAschaenz committed:
ISI-407 Bedarfsmeldungen: Footer ausblenden

lhmAschaenz committed:
ISI-407 Bugfix: isEditableBySachbearbeitung()

lhmAschaenz committed:
ISI-407 Bedingt SoBoN-Orientierungwert validieren

boal commented on frontend/src/components/abfragevarianten/AbfragevarianteSachbearbeitungFormular.vue:
```diff
@@ -0,0 +1,269 @@
+    return this.$store.getters["lookup/infrastruktureinrichtungTyp"];
+  }
+
+  private bedarfsmeldungErfassen(): void {
```
Das Verb besser vor der Entität schreiben. -> `erfassenBedarfsmeldung`

boal commented on frontend/src/components/abfragevarianten/AbfragevarianteSachbearbeitungFormular.vue:
```diff
@@ -0,0 +1,269 @@
+    this.bedarfsmeldungFachabteilungenDialogOpen = true;
+  }
+
+  private bedarfsmeldungUebernehmen(bedarfsmeldung: BedarfsmeldungFachabteilungenModel): void {
```
Das Verb besser vor der Entität schreiben. -> `uebernehmenBedarfsmeldung`

boal commented on frontend/src/components/abfragevarianten/BedarfsmeldungFachabteilungenDialog.vue:
```diff
@@ -0,0 +1,171 @@
+  }
+
+  /**
+   * Die Methode setzt die Validatierung der Dialog-Form zurück, wenn sich der Wert von showBedarfsmeldungDialog ändert.
```
Typo: Anstelle Validatierung  -> Validierung

lhmAschaenz committed:
ISI-407 Review Anmerkungen eingearbeitet

banzuu commented on frontend/src/components/abfragevarianten/AbfragevarianteSachbearbeitungFormular.vue:
```diff
@@ -0,0 +1,269 @@
+  }
+
+  private erfassenBedarfsmeldung(): void {
+    this.currentBedarfsmeldung = _.clone(createBedarfsmeldungFachabteilungenDto());
```
Ist das Klonen hier notwendig?

banzuu commented on frontend/src/components/abfragevarianten/AbfragevarianteSachbearbeitungFormular.vue:
```diff
@@ -0,0 +1,269 @@
+
+  private editBedarfsmeldung(bedarfsmeldung: BedarfsmeldungFachabteilungenModel, itemIndex: number): void {
+    this.selectedItemIndex = itemIndex;
+    this.currentBedarfsmeldung = _.clone(bedarfsmeldung);
```
Ist das Klonen hier notwendig?

lhmAschaenz replied:
entfernt.

lhmAschaenz commented on frontend/src/utils/MapperUtil.ts:
```diff
@@ -57,18 +71,98 @@ export function mapToAbfrageerstellungInfrastrukturabfrageAngelegt(
+      anzahlKindergartengruppen: bedarfmeldung.anzahlKindergartengruppen,
+      anzahlHortgruppen: bedarfmeldung.anzahlHortgruppen,
+      anzahlGrundschulzuege: bedarfmeldung.anzahlGrundschulzuege,
+    } as BedarfsmeldungFachabteilungenDto;
```
Wir würden gerne die Casts zum einen zur Erhöhung der Lesbarkeit zum anderen ist dies auch für die Typprüfung des TS-Compilers stellenweise nötig (Prüfung auf Übereinstimmung Return-Typs mit dem Typ des erstellten Objektes (as)).

lhmAschaenz commented on frontend/src/utils/MapperUtil.ts:
```diff
@@ -57,18 +71,98 @@ export function mapToAbfrageerstellungInfrastrukturabfrageAngelegt(
+    bedarfsmeldungFachreferate: mapToBedarfsmeldungFachabteilungen(
+      abfragevarianteSachbearbeitung?.bedarfsmeldungFachreferate
+    ),
+  } as AbfragevarianteSachbearbeitungDto;
```
Wir würden gerne die Casts zum einen zur Erhöhung der Lesbarkeit zum anderen ist dies auch für die Typprüfung des TS-Compilers stellenweise nötig (Prüfung auf Übereinstimmung Return-Typs mit dem Typ des erstellten Objektes (as)).

lhmAschaenz commented on frontend/src/components/abfragevarianten/AbfragevarianteSachbearbeitungFormular.vue:
```diff
@@ -0,0 +1,269 @@
+  get sobonOrientierungswertJahrValidator(): unknown[] {
+    if (this.isEditableBySachbearbeitung()) {
+      const usedRules: unknown[] = [];
+      // Da die Composition API keine Mixins unterstützt, müssen die Rules importiert werden.
```
Kommentar angepasst

lhmAschaenz committed:
ISI-407 Review Anmerkungen eingearbeitet

lhmAschaenz committed:
ISI-407 Review Anmerkungen eingearbeitet-407 Review Anmerkungen
eingearbeitet-407 Review Anmerkungen eingearbeitet-407 Review
Anmerkungen eingearbeitet-407 Review Anmerkungen eingearbeitet-407
Review Anmerkungen eingearbeitet-407 Review Anmerkungen
eingearbeitet-407 Review Anmerkungen eingearbeitet-407 Review
Anmerkungen eingearbeitet-407 Review Anmerkungen eingearbeitet-407
Review Anmerkungen eingearbeitet-407 Review Anmerkungen
eingearbeitet-407 Review Anmerkungen eingearbeitet-407 Review
Anmerkungen eingearbeitet-407 Review Anmerkungen eingearbeitet-407
Review Anmerkungen eingearbeitet-407 Review Anmerkungen
eingearbeitet-407 Review Anmerkungen eingearbeitet-407 Review
Anmerkungen eingearbeitet-407 Review Anmerkungen eingearbeitet-407
Review Anmerkungen eingearbeitet-407 Review Anmerkungen
eingearbeitet-407 Review Anmerkungen eingearbeitet-407 Review
Anmerkungen eingearbeitet-407 Review Anmerkungen eingearbeitet-407
Review Anmerkungen eingearbeitet-407 Review Anmerkungen
eingearbeitet-407 Review Anmerkungen eingearbeitet-407 Review
Anmerkungen eingearbeitet-407 Review Anmerkungen eingearbeitet-407
Review Anmerkungen eingearbeitet-407 Review Anmerkungen
eingearbeitet-407 Review Anmerkungen eingearbeitet-407 Review
Anmerkungen eingearbeitet-407 Review Anmerkungen eingearbeitet-407
Review Anmerkungen eingearbeitet-407 Review Anmerkungen
eingearbeitet-407 Review Anmerkungen eingearbeitet-407 Review
Anmerkungen eingearbeitet-407 Review Anmerkungen eingearbeitet your changes. Lines starting

lhmAschaenz committed:
ISI-407 Review Anmerkungen eingearbeitet

boal commented on frontend/src/utils/MapperUtil.ts:
```diff
@@ -33,7 +47,7 @@ export function mapToAbfrageerstellungInfrastrukturabfrageAngelegt(
       geschossflaecheWohnenGenehmigt: abfragevariante.geschossflaecheWohnenGenehmigt,
       geschossflaecheWohnenSoBoNursaechlich: abfragevariante.geschossflaecheWohnenSoBoNursaechlich,
-    } as AbfrageerstellungAbfragevarianteAngelegtDto;
+    } as AbfragevarianteAngelegtDto;
```
Cast ist erforderlich um der IDE eine korrekte Code-Validierung zu ermöglichen.

Des Weiteren hat der Cast einen Dokumentationscharakter, da bei Erstellung des Objekts dann eindeutig definiert ist, um welche Typ es sich handelt.

boal commented on frontend/src/utils/MapperUtil.ts:
```diff
@@ -48,7 +62,7 @@ export function mapToAbfrageerstellungInfrastrukturabfrageAngelegt(
       bebauungsplannummer: infrastrukturabfrageDto.abfrage?.bebauungsplannummer,
       dokumente: infrastrukturabfrageDto.abfrage?.dokumente,
-    } as AbfrageerstellungAbfrageAngelegtDto,
+    } as AbfrageAngelegtDto,
```
Cast ist erforderlich um der IDE eine korrekte Code-Validierung zu ermöglichen.

Des Weiteren hat der Cast einen Dokumentationscharakter, da bei Erstellung des Objekts dann eindeutig definiert ist, um welche Typ es sich handelt.

boal commented on frontend/src/utils/MapperUtil.ts:
```diff
@@ -57,18 +71,98 @@ export function mapToAbfrageerstellungInfrastrukturabfrageAngelegt(
+      abfragevarianteSachbearbeitung: mapToAbfragevarianteSachbearbeitungDto(
+        abfragevariante.abfragevarianteSachbearbeitung
+      ),
+    } as AbfragevarianteSachbearbeitungInBearbeitungSachbearbeitungDto;
```
Cast ist erforderlich um der IDE eine korrekte Code-Validierung zu ermöglichen.

Des Weiteren hat der Cast einen Dokumentationscharakter, da bei Erstellung des Objekts dann eindeutig definiert ist, um welche Typ es sich handelt.

boal commented on frontend/src/utils/MapperUtil.ts:
```diff
@@ -57,18 +71,98 @@ export function mapToAbfrageerstellungInfrastrukturabfrageAngelegt(
+        abfragevarianteSachbearbeitung: mapToAbfragevarianteSachbearbeitungDto(
+          abfragevariante.abfragevarianteSachbearbeitung
+        ),
+      } as AbfragevarianteInBearbeitungSachbearbeitungDto;
```
Cast ist erforderlich um der IDE eine korrekte Code-Validierung zu ermöglichen.

Des Weiteren hat der Cast einen Dokumentationscharakter, da bei Erstellung des Objekts dann eindeutig definiert ist, um welche Typ es sich handelt.

boal commented on frontend/src/utils/MapperUtil.ts:
```diff
@@ -57,18 +71,98 @@ export function mapToAbfrageerstellungInfrastrukturabfrageAngelegt(
+    version: infrastrukturabfrageDto.version,
+    abfragevarianten: abfragevarianten,
+    abfragevariantenSachbearbeitung: abfragevariantenSachbearbeitung,
+  } as InfrastrukturabfrageInBearbeitungSachbearbeitungDto;
```
Cast ist erforderlich um der IDE eine korrekte Code-Validierung zu ermöglichen.

Des Weiteren hat der Cast einen Dokumentationscharakter, da bei Erstellung des Objekts dann eindeutig definiert ist, um welche Typ es sich handelt.

boal committed:
resolve merge conflicts

boal committed:
change to iseditable

FatihLHM committed:
Bugfix formChanged Datepicker

boal committed:
change to clone deep to avoid referencing the object
